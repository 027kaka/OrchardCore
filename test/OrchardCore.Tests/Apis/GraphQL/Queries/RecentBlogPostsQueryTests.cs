using System.Threading.Tasks;
using OrchardCore.ContentManagement;
using OrchardCore.Lists.Models;
using OrchardCore.Tests.Apis.Context;
using Xunit;

namespace OrchardCore.Tests.Apis.GraphQL.Queries
{
    public class RecentBlogPostsQueryTests
    {
        [Fact]
        public async Task ShouldListBlogPostWhenCallingAQuery()
        {
            await BlogContext.InitializeBlogAsync();

            var blogPostContentItemId = await BlogContext
                .CreateContentItem("BlogPost", builder =>
                {
                    builder.Published = true;
                    builder.Latest = true;
                    builder.DisplayText = "Some sorta blogpost in a Query!";

                    builder
                        .Weld(new ContainedPart
                        {
                            ListContentItemId = BlogContext.BlogContentItemId
                        });
                });

            var result = await BlogContext
                .GraphQLClient
                .Content
                .Query("RecentBlogPosts", builder =>
                {
                    builder
                        .AddField("displayText");
                });

            var nodes = result["data"]["recentBlogPosts"];

            Assert.Equal("Some sorta blogpost in a Query!", nodes[0]["displayText"].ToString());
            Assert.Equal("Man must explore, and this is exploration at its greatest", nodes[1]["displayText"].ToString());
        }
    }
}
