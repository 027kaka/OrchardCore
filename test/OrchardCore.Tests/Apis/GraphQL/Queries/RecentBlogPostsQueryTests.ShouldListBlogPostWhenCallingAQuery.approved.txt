{
  "data": {
    "recentBlogPosts": [
      {
        "titlePart": {
          "Title": "Some sorta blogpost in a Query!"
        }
      },
      {
        "titlePart": {
          "Title": "Man must explore, and this is exploration at its greatest"
        }
      }
    ]
  }
}